homeassistant/components/proxmox/__init__.py

```diff
@@ -0,0 +1,132 @@
+"""Support for Proxmox Virtual Environment."""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME, CONF_VERIFY_SSL)
+from homeassistant.core import callback
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+REQUIREMENTS = ['proxmoxer==1.0.3']
+DOMAIN = 'proxmox'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+UPDATE_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+DATA_PROXMOX_NODES = 'proxmox_nodes'
+DATA_PROXMOX_CONTROL = 'proxmox_control'
+SIGNAL_PROXMOX_UPDATED = 'proxmox_updated'
+DEFAULT_PORT = 8006
+DEFAULT_REALM = 'pam'
+DEFAULT_VERIFY_SSL = True
+
+CONF_REALM = 'realm'
+CONF_NODES = 'nodes'
+CONF_VMS = 'vms'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_REALM, default=DEFAULT_REALM): cv.string,
+        vol.Required(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+        vol.Optional(CONF_NODES, default=[]): [cv.string],
+        vol.Optional(CONF_VMS, default=[]): [int],
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+HOST_CONFIG_SCHEMA = vol.Schema({
+})
+
+
+async def async_configure_proxmox(hass, config):
```

Comment:
Remove this and add the content to `async_setup()`.

Reply:
Moved to async_setup()